nav2_recoveries/src/spin.cpp

```diff
@@ -87,13 +88,28 @@ Status Spin::onCycleUpdate()
   geometry_msgs::msg::Pose2D pose2d;
   pose2d.x = current_pose.position.x;
   pose2d.y = current_pose.position.y;
-  pose2d.theta = tf2::getYaw(current_pose.orientation) +
-    cmd_vel.angular.z * (1 / cycle_frequency_);
+  pose2d.theta = current_yaw;
 
-  if (!collision_checker_->isCollisionFree(pose2d)) {
-    stopRobot();
-    RCLCPP_WARN(node_->get_logger(), "Collision Ahead - Exiting Spin ");
-    return Status::SUCCEEDED;
+  // Simulate ahead by simulate_ahead_time_ in cycle_frequency_ increments
+
+  int cycle_count = 0;
+  double sim_position_change;
```

Comment:
maybe move to the while loop

Reply:
so I didn't put them in the loop since its a little more efficient to not create the variable in the loop since the C++ standard destroys and reconstructs it each iteration.